numpy/lib/_shape_base_impl.py

```diff
@@ -1039,8 +1039,19 @@ def dsplit(ary, indices_or_sections):
 def get_array_prepare(*args):
     """Find the wrapper for the array with the highest priority.
 
-    In case of ties, leftmost wins. If no wrapper is found, return None
+    In case of ties, leftmost wins. If no wrapper is found, return None.
+
+    .. deprecated:: 2.0
     """
+
+    # Deprecated in NumPy 2.0, 2023-08-28
+    warnings.warn(
+        "`get_array_prepare` is deprecated. "
+        "(deprecated in NumPy 2.0)",
+        DeprecationWarning,
+        stacklevel=2
+    )
+
```

Comment:
IMO no need to deprecate it, just remove it. It has no internal consumers and as far as I can see no public consumers on github besides wrappers for the entire numpy API.

Reply:
I agree that's even better option for that one - removed.